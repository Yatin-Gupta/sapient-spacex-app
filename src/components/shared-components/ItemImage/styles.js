/**
 * ItemImage Stylesheet in JSS
 */
const imgBackgroundColor = "#f2f2f2";
const styles = {
  imgDiv: {
    backgroundColor: imgBackgroundColor,
    "&>img": {
      width: "100%",
      height: "234.42px",
    },
  },
};

export default styles;
