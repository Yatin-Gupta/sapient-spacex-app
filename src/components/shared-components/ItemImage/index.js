import React from "react";
import { createUseStyles } from "react-jss";
import styles from "./styles";

const useStyles = createUseStyles(styles);

/**
 * Image component used in every spaceX Program Item
 * @param {*} props
 */
function ItemImage(props) {
  const classes = useStyles();
  if (!props.src) {
    return <React.Fragment />;
  }
  const imgDivClassName = props.className ? `${classes.imgDiv} ${props.className}` : classes.imgDiv;
  return (
    <React.Fragment>
      <div {...props} className={imgDivClassName} role="img" aria-label={props.imgDesc ? props.imgDesc : ""}>
        <img src={props.src} alt={props.alt ? props.alt : "spacex-launch-program-image"} />
      </div>
    </React.Fragment>
  );
}

export default ItemImage;
