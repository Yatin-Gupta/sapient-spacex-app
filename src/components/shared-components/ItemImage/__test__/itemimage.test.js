import React from "react";
import ReactDOM from "react-dom";
import ItemImage from "../index";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("ItemImage component renders without crashing", () => {
  const div = document.createElement("div");
  const dummyImageURL = "https://dummyimage.com/qvga";
  ReactDOM.render(<ItemImage src={dummyImageURL} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("ItemImage component renders with src attribute passed as props", () => {
  const dummyImageURL = "https://dummyimage.com/qvga";
  const { getByTestId } = render(<ItemImage src={dummyImageURL} />);
  expect(getByTestId("item-image")).toHaveAttribute("src", dummyImageURL);
});

it("ItemImage component matches the snapshot", () => {
  const dummyImageURL = "https://dummyimage.com/qvga";
  const tree = renderer.create(<ItemImage src={dummyImageURL} />).toJSON();
  expect(tree).toMatchSnapshot();
});
