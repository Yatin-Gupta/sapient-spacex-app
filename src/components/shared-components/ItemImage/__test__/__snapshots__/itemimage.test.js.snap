// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemImage component matches the snapshot 1`] = `
<div
  aria-label=""
  className="imgDiv-0-2-1"
  role="img"
  src="https://dummyimage.com/qvga"
>
  <img
    alt="spacex-launch-program-image"
    data-testid="item-image"
    src="https://dummyimage.com/qvga"
  />
</div>
`;
